src/app/embed/adapters/feed.adapter.ts

```diff
@@ -0,0 +1,74 @@
+import { EMBED_FEED_URL_PARAM, EMBED_FEED_ID_PARAM, EMBED_EPISODE_GUID_PARAM } from './../embed.constants'
+import { Http, Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+
+export class FeedAdapter {
+	private feedUrl: string
+	private feedId: string
+	private guid: string
+
+  constructor(
+		private params: Object,
+		private http: Http
+	) {
+		this.feedUrl = params[EMBED_FEED_URL_PARAM]
+		this.feedId  = params[EMBED_FEED_ID_PARAM]
+		this.guid = params[EMBED_EPISODE_GUID_PARAM]
+  }
+
+	public get getParams(): Observable<Object> { 
+ 		return this.http.get(this.draperUrl).map(this.parseResponse)
+	}
+
+	private get draperUrl(): string { 
+		let feedUrl = `https://draper.radiopublic.com/transform?url=${this.feedUrl}`
+		let feedId;
+		
+		return (feedUrl || feedId)
+	}
+
+	private parseResponse(res: Response): Object {
```

Comment:
I think you ought to define an interface instead of using the Object return type.